cmd/backup_keys.go

```diff
@@ -158,7 +161,9 @@ func createTar(source string, target string) error {
 			if err != nil {
 				return err
 			}
-			defer file.Close()
+			defer func() {
+				err = file.Close()
```

Comment:
I think those error assignments inside the defers might shadow each other and the final io.Copy error...

also it would be a bit convoluted to get it right here I think:
- change the io.Copy to somethning like `if _, err:= io.Copy(tarball, file); err!=nil{ return err; }`
- other defer with errors should probably check if err is not nil before setting over it

it'll probably will be quite a bit verbose though, not sure if there are better ways of doing this

Reply:
Ah so true. I can change that so the error message doesn't get overwritten